cmd/monaco/integrationtest/utils/monaco/cmd.go

```diff
@@ -21,45 +23,40 @@ import (
 	"fmt"
 	"regexp"
 	"strings"
+	"testing"
 
 	"github.com/spf13/afero"
 
 	"github.com/dynatrace/dynatrace-configuration-as-code/v2/cmd/monaco/runner"
 )
 
-// The Monaco is the entry point for integration tests. It accepts a command in a way it would be called via a CLI. `monaco` keyword can be omitted from the command.
-// To execute command invoke Run() method
-//
-// monaco.Monaco("monaco download manifest.yaml --enviroment=my_env").Run()
-
-func Run(command string) error {
-	return RunWithFs(newFs(), command)
-}
-
-func Runf(command string, args ...any) error {
-	return RunWithFSf(newFs(), command, args...)
-}
+func NewTestFs() afero.Fs { return afero.NewCopyOnWriteFs(afero.NewOsFs(), afero.NewMemMapFs()) }
 
-func newFs() afero.Fs { return afero.NewCopyOnWriteFs(afero.NewOsFs(), afero.NewMemMapFs()) }
+// spacesRegex finds all sequential spaces
+var spacesRegex = regexp.MustCompile(`\s+`)
 
-func RunWithFSf(fs afero.Fs, command string, args ...any) error {
-	return RunWithFs(fs, fmt.Sprintf(command, args...))
-}
-
-func RunWithFs(fs afero.Fs, command string) error {
+// Run is the entrypoint to run monaco for all integration tests.
+// It requires to specify the full command (`monaco [deploy]....`) and sets up the runner.
+func Run(t *testing.T, fs afero.Fs, command string) error {
```

Comment:
Question: do we need to tie this function to `t *testing.T` or would `context.Context` be a better choice? The latter would require a different log line inside but nothing more (I think). Another option might be to go for `*testing.TB`, although this would be a simple change to do in a follow-up if we need it.

Reply:
Context would be enough for sure, having it in also does not hurt IMO, marking the function explicitly as 'testing' function.

Reply:
Let's leave it is as is then 👍